services/src/functional-tests/src/test/groovy/org/openkilda/functionaltests/spec/northbound/switches/SwitchSpec.groovy

```diff
@@ -24,4 +33,118 @@ class SwitchSpec extends BaseSpecification {
+
+        then: "Got 400 BadRequest because switch has ISLs"
+        def exc = thrown(HttpClientErrorException)
+        exc.rawStatusCode == 400
+
+        and: "Cleanup: Activate the switch back"
+        lockKeeperService.reviveSwitch(sw.getDpId())
+    }
+
+    def "Delete inactive switch with single flow"() {
+        requireProfiles("virtual")
+
+        given: "An active switch with flow"
+        def sw = topology.getActiveSwitches()[0]
+        def flow = flowHelper.singleSwitchFlow(sw)
+        northboundService.addFlow(flow)
+        assert Wrappers.wait(WAIT_OFFSET) { northboundService.getFlowStatus(flow.id).status == FlowState.UP }
+
+        and: "Deactivate this switch"
+        lockKeeperService.knockoutSwitch(sw.getDpId())
+
+        when: "Try to delete inactive switch with flow"
+        northboundService.deleteSwitch(sw.getDpId(), false)
+
+        then: "Got 400 BadRequest because switch has flow"
+        def exc = thrown(HttpClientErrorException)
+        exc.rawStatusCode == 400
+
+        and: "Cleanup: Activate the switch back and remove flow"
+        lockKeeperService.reviveSwitch(sw.getDpId())
+        northboundService.deleteFlow(flow.id)
+    }
+
+    def "Delete inactive switch with flow"() {
+        requireProfiles("virtual")
+
+        given: "An active switch with flow"
+        def (Switch srcSwitch, Switch dstSwitch) = topology.getActiveSwitches()[0..1]
+        def flow = flowHelper.randomFlow(srcSwitch, dstSwitch)
+        northboundService.addFlow(flow)
+        assert Wrappers.wait(WAIT_OFFSET) { northboundService.getFlowStatus(flow.id).status == FlowState.UP }
+
+        and: "Deactivate this switch"
+        lockKeeperService.knockoutSwitch(srcSwitch.getDpId())
+
+        when: "Try to delete inactive switch with flow"
+        northboundService.deleteSwitch(srcSwitch.getDpId(), false)
+
+        then: "Got 400 BadRequest because switch has flow"
+        def exc = thrown(HttpClientErrorException)
+        exc.rawStatusCode == 400
+
+        and: "Cleanup: Activate the switch back and remove flow"
+        lockKeeperService.reviveSwitch(srcSwitch.getDpId())
+        northboundService.deleteFlow(flow.id)
+    }
+
+    def "Force switch delete"() {
+        requireProfiles("virtual")
+
+        given: "An active switch"
+        def sw = topology.getActiveSwitches()[0]
+
+        when: "Try to force delete active switch with ISLs"
+        def response = northboundService.deleteSwitch(sw.getDpId(), true)
+
+        then: "Check that switch is actually deleted"
+        response.deleted
+        Wrappers.wait(WAIT_OFFSET) {
+            !northboundService.getAllSwitches().stream().anyMatch({ s -> (s.switchId == sw.dpId) })
+        }
+
+        and: "Cleanup: Restore the switch"
+        lockKeeperService.reviveSwitch(sw.getDpId())
+        Wrappers.wait(WAIT_OFFSET) {
+            northboundService.getAllSwitches().stream().anyMatch({ s -> (s.switchId == sw.dpId) })
+        }
+    }
```

Comment:
missing positive case without force delete?

Reply:
yes( I'm still trying to find a way to remove inactive ISLs from DB and restore it after

Reply:
Correct me if I'm wrong, but I think it should require the ISL to be in FAILED status (not necessary to remove it from DB entirely, just deactivate it). At least I would expect it to work this way

Reply:
Discussed with @surabujin . The switch should not have **any** ISLs before deletion (just like it is implemented now). So for now the only way is to remove an ISL directly in DB

Reply:
I rebased on Delete ISL pull request so now we can remove ISLs via API